### ISSUE TITLE
Cannot get wakewordAgent to install correctly?

### ISSUE THREAD

**Issue by KilrBe3** [alexa/alexa-avs-sample-app#630]
Ran through this two times now, once a error of chksshpwd  , which I fixed and discover is a new bug in Rasbian OS. Then I still run into this issue every time, and when opening up final terminal to start wake word agent "no file is found". As other error's have reported here. Though this to me, seems more a portaudio thing? Ring? So confused!

She works when I press listen, as others have gotten her. Just not her wake word.

Cheers.

========== Preparing External dependencies for Wake Word Agent ==========
cp: cannot stat ‘/home/pi/Desktop/alexa-avs-sample-app/samples/wakeWordAgent/kitt_ai/snowboy/examples/C++/portaudio/install/include/pa_ringbuffer.h’: No such file or directory
cp: cannot stat ‘/home/pi/Desktop/alexa-avs-sample-app/samples/wakeWordAgent/kitt_ai/snowboy/examples/C++/portaudio/install/include/pa_util.h’: No such file or directory
========== Compiling Wake Word Agent ==========
-- The C compiler identification is GNU 4.9.2
-- The CXX compiler identification is GNU 4.9.2
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Configuring done
-- Generating done
-- Build files have been written to: /home/pi/Desktop/alexa-avs-sample-app/samples/wakeWordAgent/src
Scanning dependencies of target wakeWordAgent
[ 13%] Building CXX object CMakeFiles/wakeWordAgent.dir/main.cpp.o
[ 13%] Building CXX object CMakeFiles/wakeWordAgent.dir/WakeWordAgent.cpp.o
[ 20%] Building CXX object CMakeFiles/wakeWordAgent.dir/WakeWordEngineFactory.cpp.o
[ 26%] Building CXX object CMakeFiles/wakeWordAgent.dir/WakeWordEngine.cpp.o
In file included from /home/pi/Desktop/alexa-avs-sample-app/samples/wakeWordAgent/src/KittAiSnowboyWakeWordEngine.h:19:0,
                 from /home/pi/Desktop/alexa-avs-sample-app/samples/wakeWordAgent/src/WakeWordEngineFactory.cpp:17:
/home/pi/Desktop/alexa-avs-sample-app/samples/wakeWordAgent/src/PortAudioWrapper.h:17:27: fatal error: pa_ringbuffer.h: No such file or directory
 #include "pa_ringbuffer.h"
                           ^
compilation terminated.
[ 33%] Building CXX object CMakeFiles/wakeWordAgent.dir/WakeWordIPCFactory.cpp.o
CMakeFiles/wakeWordAgent.dir/build.make:110: recipe for target 'CMakeFiles/wakeWordAgent.dir/WakeWordEngineFactory.cpp.o' failed
make[2]: *** [CMakeFiles/wakeWordAgent.dir/WakeWordEngineFactory.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
CMakeFiles/Makefile2:67: recipe for target 'CMakeFiles/wakeWordAgent.dir/all' failed
make[1]: *** [CMakeFiles/wakeWordAgent.dir/all] Error 2
Makefile:83: recipe for target 'all' failed
make: *** [all] Error 2
-- The C compiler identification is GNU 4.9.2
-- The CXX compiler identification is GNU 4.9.2
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Configuring done
-- Generating done
-- Build files have been written to: /home/pi/Desktop/alexa-avs-sample-app/samples/wakeWordAgent/tst
Scanning dependencies of target wakeWordAgentTest
[  5%] Building CXX object CMakeFiles/wakeWordAgentTest.dir/main.cpp.o
[ 10%] Building CXX object CMakeFiles/wakeWordAgentTest.dir/testKittAi.cpp.o
[ 15%] Building CXX object CMakeFiles/wakeWordAgentTest.dir/testLogger.cpp.o
[ 20%] Building CXX object CMakeFiles/wakeWordAgentTest.dir/testIPCFactory.cpp.o
In file included from /home/pi/Desktop/alexa-avs-sample-app/samples/wakeWordAgent/tst/../src/KittAiSnowboyWakeWordEngine.h:19:0,
                 from /home/pi/Desktop/alexa-avs-sample-app/samples/wakeWordAgent/tst/testKittAi.cpp:15:
/home/pi/Desktop/alexa-avs-sample-app/samples/wakeWordAgent/tst/../src/PortAudioWrapper.h:17:27: fatal error: pa_ringbuffer.h: No such file or directory
 #include "pa_ringbuffer.h"
                           ^
compilation terminated.
[ 25%] Building CXX object CMakeFiles/wakeWordAgentTest.dir/testSensory.cpp.o
CMakeFiles/wakeWordAgentTest.dir/build.make:110: recipe for target 'CMakeFiles/wakeWordAgentTest.dir/testKittAi.cpp.o' failed
make[2]: *** [CMakeFiles/wakeWordAgentTest.dir/testKittAi.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
CMakeFiles/Makefile2:67: recipe for target 'CMakeFiles/wakeWordAgentTest.dir/all' failed
make[1]: *** [CMakeFiles/wakeWordAgentTest.dir/all] Error 2
Makefile:83: recipe for target 'all' failed
make: *** [all] Error 2

=============================
*****************************
========= Finished ==========
*****************************
=============================

**Comment by KilrBe3** [comment]
EDIT: 
Fixed doing this:

What I did was to go to:
/alexa-avs-sample-app/samples/wakeWordAgent/kitt_ai/snowboy/examples/C++/portaudio/src/common
copy all the .h files you need to make the wakeWordAgent (for me these were pa_ringbuffer.h and pa_util.h) to:
alexa-avs-sample-app/samples/wakeWordAgent/ext/include
and then go back to:
alexa-avs-sample-app/samples/wakeWordAgent/src
to run make.

Fixed. Did end up having to exactly add both pa_ring and pa_util. Once both those copied. Re-run automated installer. Boom, She works!

So I guess then, this is a problem with the Alexa Team and the automated installer setup. Needs official looking at IMHO :)

**Comment by zomzom96** [comment]
Hello,

I've managed to track down the cause of the service not working, to be that I'm missing the dependencies that are listed in the automated.sh (e.g. pa_ringbuffer.h ....etc.) I looked at this fix here and I can see that @KilrBe3 Solved through going to the directory under ...C++/portaudio/src/common. However, this directory didn't exist because there was a problem when install port audio script was run. I managed to download the tgz file and install port audio from www.portaudio.com . Then, I ran these scripts inside the file after gunzip and un-tar it. (./configure ---> make  ---> make test ---> sudo make install ) however that all worked fine. Still though can't find the directory that contains all the .h extension files... any help would be greatly appreciated. Thanks!
